hastory-data/src/Data/Hastory/Utils.hs

```diff
@@ -20,3 +30,33 @@ doCountsWith conv func = foldl go HM.empty
       where
         a Nothing = Just 1
         a (Just d) = Just $ d + func k
+
+
+-- |Clients are expected to write their tests in such a way as to avoid
```

Comment:
This is a really bad idea.
It's best to enforce this using `around` than to require it with `beforeAll`.

Reply:
Then you can use the `with` function which calls `close` correctly.